Makefile

```diff
@@ -6,51 +6,53 @@ scripts_dir = scripts
 appengine_dir = dist/gae
 appengine_scripts_dir = $(appengine_dir)/scripts
 
-
+z3_dir = third_party/z3
+python_prefix = $(shell python2.7 -c "import sys; print sys.prefix")
+# Super hacky!
+python_package_dir = $(python_prefix)/lib/python2.7/site-packages
 PYTHON_PACKAGES=networkx unittest2 werkzeug webapp2 webob jinja2
-# Prod packages: cherrypy
-
-PYTHON_EGGS=$(patsubst %,sayc-env/%.STAMP,$(PYTHON_PACKAGES))
 
 all:
+	@echo "Run 'make deps' to bring your dependencies up-to-date."
 	@echo "Run 'make check' to check against the current baseline and 'make accept' to set a new baseline from the last results."
 
-sayc-env:
-	@virtualenv sayc-env
+deps: z3py_install
+	@pip install --upgrade $(PYTHON_PACKAGES)
+
+$(z3_dir)/build/config.mk:
+	@cd $(z3_dir) && python scripts/mk_make.py --prefix=$(python_prefix)
```

Comment:
No the Makefile will do an in-tree build of z3, lib and py components, and drop them in the right place. Well, the version I have locally will ;-) It needed a small tweak.